Source/WebCore/page/scrolling/AsyncScrollingCoordinator.cpp

```diff
@@ -246,11 +282,11 @@ void AsyncScrollingCoordinator::frameViewRootLayerDidChange(LocalFrameView& fram
     
     // If the root layer does not have a ScrollingStateNode, then we should create one.
     ensureRootStateNodeForFrameView(frameView);
-    ASSERT(m_scrollingStateTree->stateNodeForID(frameView.scrollingNodeID()));
+    ASSERT(ensureScrollingStateTreeForRootFrameID(frameView.frame().rootFrame().frameID()).stateNodeForID(frameView.scrollingNodeID()));
```

Comment:
It's bad to call an `ensure` function inside an assertion, because you've now got a debug-only behavior difference.

Reply:
Removed.